kongswap_adaptor/tests/allowance_expiry.rs

```diff
@@ -0,0 +1,386 @@
+mod helpers;
+mod pocket_ic_agent;
+
+use candid::{Nat, Principal};
+use helpers::Wasm;
+use ic_icrc1_ledger::{InitArgsBuilder, LedgerArgument};
+use ic_management_canister_types::CanisterSettings;
+use icp_ledger::{AccountIdentifier, LedgerCanisterInitPayload};
+use icrc_ledger_types::{
+    icrc1::{account::Account, transfer::TransferArg},
+    icrc2::{approve::ApproveArgs, transfer_from::TransferFromArgs},
+};
+use kongswap_adaptor::agent::AbstractAgent;
+use lazy_static::lazy_static;
+use pocket_ic::{nonblocking::PocketIc, PocketIcBuilder};
+use pocket_ic_agent::PocketIcAgent;
+// use pretty_assertions::assert_eq;
+use sha2::Digest;
+use std::time::Duration;
+
+pub const STARTING_CYCLES_PER_CANISTER: u128 = 2_000_000_000_000_000;
+
+const E8: u64 = 100_000_000;
+const NS_IN_SECOND: u64 = 1_000_000_000;
+const ONE_HOUR: u64 = 60 * 60 * NS_IN_SECOND;
+
+lazy_static! {
+    static ref SNS_LEDGER_CANISTER_ID: Principal =
+        Principal::from_text("jg2ra-syaaa-aaaaq-aaewa-cai").unwrap();
+    static ref SNS_ROOT_CANISTER_ID: Principal =
+        Principal::from_text("ju4gz-6iaaa-aaaaq-aaeva-cai").unwrap();
+    static ref NNS_ROOT_CANISTER_ID: Principal =
+        Principal::from_text("r7inp-6aaaa-aaaaa-aaabq-cai").unwrap();
+    static ref SNS_GOVERNANCE_CANISTER_ID: Principal =
+        Principal::from_text("jt5an-tqaaa-aaaaq-aaevq-cai").unwrap();
+    static ref ICP_LEDGER_CANISTER_ID: Principal =
+        Principal::from_text("ryjl3-tyaaa-aaaaa-aaaba-cai").unwrap();
+    static ref NNS_GOVERNANCE_CANISTER_ID: Principal =
+        Principal::from_text("rrkah-fqaaa-aaaaa-aaaaq-cai").unwrap();
+}
+
+#[tokio::test]
+async fn allowance_expiry() {
+    // Prepare the world.
+
+    let pocket_ic = PocketIcBuilder::new()
+        .with_nns_subnet()
+        .with_sns_subnet()
+        .with_fiduciary_subnet()
+        .build_async()
+        .await;
+
+    let mut agent = PocketIcAgent::new(pocket_ic);
+
+    let sns_ledger_canister_ic = install_sns_ledger(&agent.pic()).await;
+    let icp_ledger_canister_id = install_icp_ledger(&agent.pic()).await;
+
+    let owner = Principal::from_text("uz6mw-eaaaa-aaaaq-aabpa-cai").unwrap();
+    let reciever = Principal::from_text("4bli7-7iaaa-aaaap-ahd4a-cai").unwrap();
+
+    mint_tokens(
+        agent.with_sender(*SNS_ROOT_CANISTER_ID),
```

Comment:
I'm confused - can SNS Root also mint tokens?  I was under the impression only SNS governance can do this

Reply:
Resolved confusion - this is from the install_sns_ledger function, that sets sns root as minting account.  Should be fixed to make SNS Gov the minting account so that this is accurate documentation.